controller/handler/handler.go

```diff
@@ -3,43 +3,49 @@ package handler
 import (
 	"net/http"
 
-	"github.com/TwiN/gatus/v4/config/ui"
-	"github.com/TwiN/gatus/v4/security"
+    "github.com/TwiN/gatus/v4/config"
 	"github.com/TwiN/health"
 	"github.com/gorilla/mux"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
-func CreateRouter(staticFolder string, securityConfig *security.Config, uiConfig *ui.Config, enabledMetrics bool) *mux.Router {
+func CreateRouter(staticFolder string, config *config.Config) *mux.Router {
 	router := mux.NewRouter()
-	if enabledMetrics {
-		router.Handle("/metrics", promhttp.Handler()).Methods("GET")
-	}
 	api := router.PathPrefix("/api").Subrouter()
 	protected := api.PathPrefix("/").Subrouter()
 	unprotected := api.PathPrefix("/").Subrouter()
-	if securityConfig != nil {
-		if err := securityConfig.RegisterHandlers(router); err != nil {
-			panic(err)
+	if config != nil {
+		if config.Metrics {
+			router.Handle("/metrics", promhttp.Handler()).Methods("GET")
 		}
-		if err := securityConfig.ApplySecurityMiddleware(protected); err != nil {
-			panic(err)
+		if config.Security != nil {
+			if err := config.Security.RegisterHandlers(router); err != nil {
+				panic(err)
+			}
+			if err := config.Security.ApplySecurityMiddleware(protected); err != nil {
+				panic(err)
+			}
 		}
+
+		// Endpoints
+
+		unprotected.Handle("/v1/config", ConfigHandler{securityConfig: config.Security}).Methods("GET")
+		unprotected.HandleFunc("/v1/endpoints/{key}/response-times/{duration}/badge.svg", ResponseTimeBadge(config)).Methods("GET")
+		unprotected.HandleFunc("/v1/services/{key}/response-times/{duration}/badge.svg", ResponseTimeBadge(config)).Methods("GET")
+		// SPA
+		router.HandleFunc("/services/{name}", SinglePageApplication(staticFolder, config.UI)).Methods("GET") // XXX: Remove this in v4.0.0
```

Comment:
There's some strange conflict going on here.. This line should no longer exist.
See https://github.com/TwiN/gatus/blob/7daf2b5cac28201c33fc924a7533609f0eb6ee8e/controller/handler/handler.go#L41